specification/deviceregistry/resource-manager/Microsoft.DeviceRegistry/preview/2023-11-01-preview/asset.json

```diff
@@ -0,0 +1,663 @@
+        }
+      },
+      "description": "Asset definition for Patch operation."
+    },
+    "AssetProperties": {
+      "description": "Defines the asset properties.",
+      "type": "object",
+      "properties": {
+        "uuid": {
+          "type": "string",
+          "description": "Globally unique, immutable, non-reusable id.",
+          "readOnly": true
+        },
+        "assetType": {
+          "type": "string",
+          "description": "Resource path to asset type (model) definition."
+        },
+        "enabled": {
+          "type": "boolean",
+          "description": "Enabled/Disabled status of the asset."
+        },
+        "externalAssetId": {
+          "type": "string",
+          "x-ms-mutability": [
+            "read",
+            "create"
+          ],
+          "description": "Asset id provided by the customer."
+        },
+        "displayName": {
+          "type": "string",
+          "description": "Human-readable display name."
+        },
+        "description": {
+          "type": "string",
+          "description": "Human-readable description of the asset."
+        },
+        "assetEndpointProfileUri": {
+          "type": "string",
+          "x-ms-mutability": [
+            "read",
+            "create"
+          ],
+          "description": "A reference to the asset endpoint profile (connection information) used by brokers to connect to an endpoint that provides data points for this asset. Must have the format <ModuleCR.metadata.namespace>/<ModuleCR.metadata.name>."
+        },
+        "version": {
+          "type": "integer",
+          "format": "int32",
+          "description": "An integer that is incremented each time the resource is modified.",
+          "readOnly": true
+        },
+        "manufacturer": {
+          "type": "string",
+          "description": "Asset manufacturer name."
+        },
+        "manufacturerUri": {
+          "type": "string",
+          "description": "Asset manufacturer URI."
+        },
+        "model": {
+          "type": "string",
+          "description": "Asset model name."
+        },
+        "productCode": {
+          "type": "string",
+          "description": "Asset product code."
+        },
+        "hardwareRevision": {
+          "type": "string",
+          "description": "Revision number of the hardware."
+        },
+        "softwareRevision": {
+          "type": "string",
+          "description": "Revision number of the software."
+        },
+        "documentationUri": {
+          "type": "string",
+          "description": "Reference to the documentation."
+        },
+        "serialNumber": {
+          "type": "string",
+          "description": "Asset serial number."
+        },
+        "attributes": {
+          "type": "object",
+          "description": "A set of key-value pairs that contain custom attributes set by the customer.",
+          "additionalProperties": true
```

Comment:
>"additionalProperties": true [](http://example.com/codeflow?start=10&length=28)

Is there any interpreation / validation happening in the control plane for the kes or values for this?

Reply:
We don't have any validation checks on the keys and values. These are modeled as open attributes as customers would like to model them as attributes on the ARM resource and have it synchronized down the edge, via ARC as attributes available on the k8 resource on the edge.